[MRG + 1] deprecated randomized_l1

Sentient07 opened the pull request:
This PR attempts to fix #8995

Sentient07 committed:
deprecated randomized_l1

Removed the deprecated module's code from examples

Added warning

removed from narrative documentation

fixed flake8

Changed deprecation style to GMM

Added tests and fixed flake8

Added removed code and updated whats_new

Modified whatsnew

Sentient07 commented:
@agramfort I'm not sure how the deprecation should be done, I added warning at the top of the file that is deprecated. @GaelVaroquaux asked me a gist of the code that has been removed, but I didn't remove any for now, rather in the warning message, said it would be removed in the next release

agramfort commented:
@Sentient07 read this http://scikit-learn.org/stable/developers/contributing.html#deprecation

you'll also need to remove reference to this code in narrative documentation

jnothman commented:
see cross_validation for another case of module-level deprecation

On 7 Jun 2017 5:59 pm, "Alexandre Gramfort" <notifications@github.com>
wrote:

> @Sentient07 <https://github.com/sentient07> read this
> http://scikit-learn.org/stable/developers/contributing.html#deprecation
>
> you'll also need to remove reference to this code in narrative
> documentation
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/9031#issuecomment-306719802>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz64zx1fUkMt-z0S65sFRGxoUoyLt5ks5sBlhogaJpZM4NyVKr>
> .
>

Sentient07 commented:
Sorry, had a long day of work and travel couldn't get back earlier. I had removed the deprecated module's code from the example section and specified the versions in which this module will be deprecated and removed.

agramfort commented:
@Sentient07 there should be no warning when running the test of building the doc

agramfort commented:
since you won't remove the tests now it means you need to capture the warnings

Sentient07 commented:
Understood. But is it preferred to remove the tests ? ( cross_validation still seems to have its test, but raising warnings when the module is trying to be imported)

Sentient07 commented:
Also, should I get rid of the `examples/linear_model/plot_sparse_recovery` module? I am not sure if it would be of much use once we get rid of the `RandomisedLasso`

agramfort commented:
keep the tests but capture warning

and remove examples and narrative doc about it

Sentient07 commented:
@agramfort I have caught and ignored the warnings in test. I test with `no-capture` option and I don't see any warning. Also, I have removed the example and the related documentation. (I am not very sure what is narrative documentation)

jnothman commented:
narrative documentation is basically anything in doc/

On 8 Jun 2017 11:45 pm, "Ramana Subramanyam" <notifications@github.com>
wrote:

@agramfort <https://github.com/agramfort> I have caught and ignored the
warnings in test. I test with no-capture option and I don't see any
warning. Also, I have removed the example and the related documentation. (I
am not very sure what is narrative documentation)

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/scikit-learn/scikit-learn/pull/9031#issuecomment-307107946>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AAEz6zI2ONvSN0fEovb7sez-OPY-hb6Eks5sB_r4gaJpZM4NyVKr>
.

jnothman commented:
flake8 failures:
```
Running flake8 on the diff in the range 623ee42..158cda7 (3 commit(s)):
--------------------------------------------------------------------------------
./sklearn/linear_model/randomized_l1.py:34:16: E127 continuation line over-indented for visual indent
               DeprecationWarning)
               ^
./sklearn/linear_model/tests/test_randomized_l1.py:20:80: E501 line too long (81 > 79 characters)
                                                    RandomizedLogisticRegression)
                                                                               ^
```

Sentient07 commented:
Sorry about the flake8 errors, I have fixed that and squashed the commits into one.

jnothman commented:
It seems I misled you. We shouldn't be doing a module deprecation like for
cross_validation. That's done where people import the model directly. We
should be deprecating each of the public objects in the module, like at
sklearn/mixture/gmm.py. Apologies.

On 15 June 2017 at 09:38, Ramana Subramanyam <notifications@github.com>
wrote:

> Sorry about the flake8 errors, I have fixed that and squashed the commits
> into one.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/9031#issuecomment-308587971>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEz6-NHJP2QcoghIufVrn_4SjdG1deTks5sEG7mgaJpZM4NyVKr>
> .
>

Sentient07 commented:
No problem, I will do the changes tomorrow.

amueller commented:
@Sentient07 are you still on this? Usually I wouldn't be nagging, but we're trying to move towards a release ;)

Sentient07 commented:
Will send a PR by tonight! Sorry, was having a deadline and couldn't find
time before :(

On 19 Jun 2017 9:38 pm, "Andreas Mueller" <notifications@github.com> wrote:

> @Sentient07 <https://github.com/sentient07> are you still on this?
> Usually I wouldn't be nagging, but we're trying to move towards a release ;)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/9031#issuecomment-309487538>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF6IcFqIzCI83YR2bSd1ebGPnzODCkWoks5sFpzwgaJpZM4NyVKr>
> .
>

Sentient07 commented:
@amueller sorry for the delay, I made the deprecation similar to GMM, fixed the merge conflict.

amueller commented on sklearn/linear_model/__init__.py:
```diff
@@ -30,8 +30,12 @@
-from .randomized_l1 import (RandomizedLasso, RandomizedLogisticRegression,
-                            lasso_stability_path)
+
+with warnings.catch_warnings():
```
you don't need to catch here, importing will not raise a deprecation warning

Sentient07 replied:
Ah! Forgot to remove

amueller reviewed (commented):
Looks good, but you need to fix the flake8 errors for travis to pass. Also, can you please add a test that checks that a warning is actually raised when instantiating an object of the class with ``assert_warns_message``? Thanks!

amueller commented:
Oh also please add an entry to whats_new.rst

Sentient07 commented:
@amueller I made the changes that you suggested. I did not test it locally as after the recent rebase, I am getting stuck with this error, "ImportError: cannot import name bincount". Google results suggest me to update numpy but that did not work for me. I am pushing leaving the tests to Travis as I don't want this to postpone the release. Apologies.

jnothman commented on sklearn/linear_model/__init__.py:
```diff
@@ -30,8 +30,7 @@
 from .passive_aggressive import PassiveAggressiveClassifier
 from .passive_aggressive import PassiveAggressiveRegressor
 from .perceptron import Perceptron
-from .randomized_l1 import (RandomizedLasso, RandomizedLogisticRegression,
```
We still need this. It is the source of test failures.

jnothman reviewed (commented):
And if you otherwise think your work here is complete and want a full review, change the title's WIP to MRG.

massich commented:
@Sentient07, this is a blocker issue. 
I think there is not much to be done in order to finish it. If you don't have the time to fix it soon, ping me and I'll take over.

Sentient07 commented:
I had pushed a commit before and it didn't get updated here. I have pushed it again. Once Travis passes, I'll change it to `mrg`

amueller commented on doc/whats_new.rst:
```diff
@@ -1279,6 +1279,9 @@ Model evaluation and meta-estimators
      :issue:`6660` by `Raghav RV`_.
 
+   - The :mod:`sklearn.linear_model.randomized_l1` is deprecated.
+     :issue: `8995` by `Andreas Mueller`_.
```
The "by" refers to the contributor, so you. You can use the ``:user:`` pattern as elsewhere in whatsnew.

amueller replied:
There should be a list of deprecated functions in 0.19 in the whatsnew.

amueller commented on sklearn/linear_model/__init__.py:
```diff
@@ -30,8 +30,10 @@
 from .passive_aggressive import PassiveAggressiveClassifier
 from .passive_aggressive import PassiveAggressiveRegressor
 from .perceptron import Perceptron
+
```
why the change?

Sentient07 replied:
Probably when I added code for catching the exception at the time of import, after removing it, this empty line came up

amueller commented on sklearn/linear_model/randomized_l1.py:
```diff
@@ -58,6 +60,8 @@ def _resample_model(estimator_func, X, y, scaling=.5, n_resampling=200,
     return scores_
 
 
+@deprecated("The class BaseRandomizedLinearModel is deprecated in 0.19"
```
Hm this should not have been public, but given that it was, deprecating it is the right thing to do.

amueller reviewed (commented):
LGTM apart from whatsnew nitpicks.

ogrisel reviewed (approved):
LGTM as well once @amueller's comments are addressed.

Sentient07 commented:
Addressed comments, squashed into one commit

ogrisel commented:
The CI errors are all cause by mldata.org 500 errors. Merging.

ogrisel commented:
Thanks @Sentient07!
